### PR TITLE
chore: refactored redis and added span length asserts

### PR DESCRIPTION
**Pull request by abhilash-sivan**

### CODE REVIEW

**Review by kirrg001 — APPROVED** [review]

File: packages/collector/test/tracing/database/redis/test.js [hunk 1]
```diff
@@ -235,6 +237,13 @@ const globalAgent = require('../../../globalAgent');
                       ]);
 
                       verifyHttpExit(controls, spans, writeEntrySpan);
+
+                      // TODO: not sure what is the difference in span length with setupType
```
Comment (aryamohanan) [inline packages/collector/test/tracing/database/redis/test.js:241]: Why?

Comment (abhilash-sivan) [reply packages/collector/test/tracing/database/redis/test.js:241]: This is from the main. we have to ignore it for now. Might change after the refactoring done in opt-in feature

Comment (aryamohanan) [reply packages/collector/test/tracing/database/redis/test.js:241]: See https://github.com/instana/nodejs/blob/main/packages/collector/test/tracing/database/redis/test.js#L31

File: packages/collector/test/tracing/database/redis/test.js [hunk 2]
```diff
@@ -235,6 +237,13 @@ const globalAgent = require('../../../globalAgent');
...
+                      if (setupType === 'cluster') {
+                        expect(spans.length).to.be.eql(2);
```
Comment (aryamohanan) [inline packages/collector/test/tracing/database/redis/test.js:243]: Why the difference in span length for cluster and standalone?

Comment (kirrg001) [reply packages/collector/test/tracing/database/redis/test.js:243]: The PR only adds span lengths for the "current state".

Comment (kirrg001) [reply packages/collector/test/tracing/database/redis/test.js:243]: We have a similar behavior for `ioredis`.

File: packages/collector/test/tracing/database/redis/test.js (hunk 1 above)
Comment (aryamohanan) [reply packages/collector/test/tracing/database/redis/test.js]: Ah I see, the initial comment was confusing.

File: packages/collector/test/tracing/database/redis/test.js (hunk 2 above)
Comment (kirrg001) [reply packages/collector/test/tracing/database/redis/test.js:243]: Extended https://jsw.ibm.com/browse/INSTA-14540

Comment (aryamohanan) [reply packages/collector/test/tracing/database/redis/test.js:243]: I see, the initial comment was confusing.

**Review by aryamohanan — APPROVED** [review]

**Pull request merged**